iree/tools/iree-run-module-vulkan-gui-main.cc

```diff
@@ -0,0 +1,461 @@
+  iree_hal_driver_t* iree_vk_driver = nullptr;
+  iree_hal_vulkan_driver_options_t options;
+  options.api_version = VK_API_VERSION_1_0;
+  options.features = static_cast<iree_hal_vulkan_features_t>(
+      IREE_HAL_VULKAN_ENABLE_DEBUG_UTILS |
+      IREE_HAL_VULKAN_ENABLE_PUSH_DESCRIPTORS);
+  IREE_CHECK_OK(iree_hal_vulkan_driver_create_using_instance(
+      options, iree_vk_syms, g_Instance, &iree_vk_driver));
+  // Create a device sharing our VkDevice and queue.
+  // We could also create a separate (possibly low priority) compute queue for
+  // IREE, and/or provide a dedicated transfer queue.
+  iree_hal_vulkan_queue_set_t compute_queue_set;
+  compute_queue_set.queue_family_index = g_QueueFamily;
+  compute_queue_set.queue_indices = 1 << 0;
+  iree_hal_vulkan_queue_set_t transfer_queue_set;
+  transfer_queue_set.queue_indices = 0;
+  iree_hal_device_t* iree_vk_device = nullptr;
+  IREE_CHECK_OK(iree_hal_vulkan_driver_wrap_device(
+      iree_vk_driver, g_PhysicalDevice, g_Device, compute_queue_set,
+      transfer_queue_set, &iree_vk_device));
+  // Create a HAL module using the HAL device.
+  iree_vm_module_t* hal_module = nullptr;
+  IREE_CHECK_OK(iree_hal_module_create(iree_vk_device, iree_allocator_system(),
+                                       &hal_module));
+
+  // Load bytecode module from embedded data.
+  IREE_LOG(INFO) << "Loading IREE byecode module...";
+  auto module_file_or = iree::GetModuleContentsFromFlags();
+  if (!module_file_or) {
+    IREE_LOG(FATAL) << "Error when reading module file"
+                    << module_file_or.status();
+  }
+  iree_vm_module_t* bytecode_module = nullptr;
+  IREE_CHECK_OK(iree_vm_bytecode_module_create(
+      iree_const_byte_span_t{
+          reinterpret_cast<const uint8_t*>(module_file_or->data()),
+          module_file_or->size()},
+      iree_allocator_null(), iree_allocator_system(), &bytecode_module));
+  // Query for details about what is in the loaded module.
+  iree_vm_module_signature_t bytecode_module_signature =
+      iree_vm_module_signature(bytecode_module);
+  IREE_LOG(INFO) << "Module loaded, have <"
+                 << bytecode_module_signature.export_function_count
+                 << "> exported functions:";
+  for (int i = 0; i < bytecode_module_signature.export_function_count; ++i) {
+    iree_string_view_t function_name;
+    iree_vm_function_signature_t function_signature;
+    IREE_CHECK_OK(bytecode_module->get_function(
+        bytecode_module->self, IREE_VM_FUNCTION_LINKAGE_EXPORT, i,
+        /*out_function=*/nullptr, &function_name, &function_signature));
+    IREE_LOG(INFO) << "  " << i << ": '"
+                   << std::string(function_name.data, function_name.size)
+                   << "' with calling convention '"
+                   << std::string(function_signature.calling_convention.data,
+                                  function_signature.calling_convention.size)
+                   << "'";
+  }
```

Comment:
Don't need to query and log all these details for this particular tool?
The Vulkan sample this is forked from is showing how to use IREE's APIs, this program is just meant to run a module?

Reply:
Good point. Removed. Still kept minimal logging to keep us informed regarding internal process.